### ISSUE TITLE
Better error message for formal parameter identifiers used in dependent types

### ISSUE THREAD

**Issue by mernst** [typetools/checker-framework#1014]
Consider the following code.

```java
import java.util.HashMap;
import java.util.Map;
import org.checkerframework.checker.nullness.qual.KeyFor;

public class ParameterName {

  void mapGet(Map<String, Object> m, @KeyFor("m") String k) {
    System.out.println(m.get(k));
  }

  static class HasMap {
    Map<String, Object> f = new HashMap<String, Object>();
  }

  void mapGet(HasMap hm, @KeyFor("hm.f") String k) {
    System.out.println(hm.f.get(k));
  }

}
```

`@KeyFor("m")` should be `@KeyFor("#1")`, and `@KeyFor("hm.f")` should be `@KeyFor("#1.f")`.  The Checker Framework does issue a warning message for each problem:

```
ParameterName.java:7: error: [expression.unparsable.type.invalid] Expression annotation invalid: [error for expression: m error: m: identifier not found]
  void mapGet(Map<String, Object> m, @KeyFor("m") String k) {
                                     ^
ParameterName.java:15: error: [expression.unparsable.type.invalid] Expression annotation invalid: [error for expression: hm.f error: hm: identifier not found]
  void mapGet(HasMap hm, @KeyFor("hm.f") String k) {
                         ^
```

However, the error messages are confusing to someone who doesn't know the syntax of [Java expressions as annotation arguments](https://checkerframework.org/manual/#java-expressions-as-arguments).

It would be better if the error message recognized when a formal parameter name is being used, and it gave an explicit suggestion to replace it by syntax such as `#1` (using the appropriate number).
